samples/snippets/src/main/java/com/example/dialogflow/ConversationProfileManagement.java

```diff
@@ -0,0 +1,137 @@
+import com.google.cloud.dialogflow.v2.HumanAgentAssistantConfig.SuggestionTriggerSettings;
+import com.google.cloud.dialogflow.v2.KnowledgeBaseName;
+import com.google.cloud.dialogflow.v2.LocationName;
+import com.google.cloud.dialogflow.v2.SuggestionFeature;
+import com.google.cloud.dialogflow.v2.SuggestionFeature.Type;
+import java.io.IOException;
+import java.util.Optional;
+
+public class ConversationProfileManagement {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project-id";
+    String location = "my-location";
+    String conversationProfileDisplayName = "my-conversation-profile-display-name";
+
+    // Create a conversation profile
+    String articleSuggestionKnowledgeBaseId = "my-fake-article-suggestion-knowledge-base-id";
+    String faqKnowledgeBaseId = "my-fake-faq-knowledge-base-id";
+    createConversationProfileArticleFaq(
+        projectId,
+        conversationProfileDisplayName,
+        location,
+        Optional.of(articleSuggestionKnowledgeBaseId),
+        Optional.of(faqKnowledgeBaseId));
+  }
+
+  // Create a conversation profile with given values about article suggestion or faq
+  public static ConversationProfile createConversationProfileArticleFaq(
+      String projectId,
+      String displayName,
+      String location,
+      Optional<String> articleSuggestionKnowledgeBaseId,
+      Optional<String> faqKnowledgeBaseId)
+      throws ApiException, IOException {
+    try (ConversationProfilesClient conversationProfilesClient =
+        ConversationProfilesClient.create()) {
+      SuggestionConfig.Builder suggestionConfigBuilder = SuggestionConfig.newBuilder();
+
+      // Add knowledge base for article suggestion
+      if (articleSuggestionKnowledgeBaseId.isPresent()) {
+        KnowledgeBaseName articleSuggestionKbName =
+            KnowledgeBaseName.of(projectId, articleSuggestionKnowledgeBaseId.get());
+        SuggestionFeatureConfig articleSuggestionFeatureConfig =
+            SuggestionFeatureConfig.newBuilder()
+                .setSuggestionFeature(
+                    SuggestionFeature.newBuilder().setType(Type.ARTICLE_SUGGESTION).build())
+                .setSuggestionTriggerSettings(
+                    SuggestionTriggerSettings.newBuilder()
+                        .setNoSmalltalk(true)
+                        .setOnlyEndUser(true)
+                        .build())
+                .setQueryConfig(
+                    SuggestionQueryConfig.newBuilder()
+                        .setKnowledgeBaseQuerySource(
+                            KnowledgeBaseQuerySource.newBuilder()
+                                .addKnowledgeBases(articleSuggestionKbName.toString()))
+                        .setMaxResults(3)
+                        .build())
+                .build();
```

Comment:
Wow - that looks complicated.  Is there an easier way to present this since we are trying to teach users here?

Reply:
Functionality-wise, it's hard to simplify it as we should include such nested configurations to make the feature work, but we can improve the readability by adding variables one by one and also add more descriptions.